src/Builder/Query/BitsAllClearOperator.php

```diff
@@ -28,16 +25,21 @@ class BitsAllClearOperator implements FieldQueryInterface, OperatorInterface
 {
     public const ENCODE = Encode::Single;
 
-    /** @var BSONArray|Binary|PackedArray|array|int|non-empty-string $bitmask */
-    public readonly Binary|PackedArray|BSONArray|array|int|string $bitmask;
+    /** @var list<Binary|int|non-empty-string> $bitmask */
+    public readonly array $bitmask;
 
     /**
-     * @param BSONArray|Binary|PackedArray|array|int|non-empty-string $bitmask
+     * @param Binary|int|non-empty-string ...$bitmask
+     * @no-named-arguments
      */
-    public function __construct(Binary|PackedArray|BSONArray|array|int|string $bitmask)
+    public function __construct(Binary|int|string ...$bitmask)
```

Comment:
Re:

 * https://github.com/mongodb/mongo-php-builder/pull/37/files#r1459574672
 * https://github.com/mongodb/mongo-php-builder/pull/37/files#r1459323536

I think the problem I'm pointing out in the above threads is due to ambiguity when a single integer is passed here. If `5` is passed to this method, it is interpreted as `[5]`, which is not the same.

I think this needs to be reworked to accept either:

 * An array of ints
 * A single int
 * A single BSON Binary

Reply:
You're right, there are 3 distinct behaviors. I'll remove the variadic and change the type to `int|binary|list<int>`.